### PR TITLE
Reorganization of the ThetaGPU tutorials

### PR DESCRIPTION
**Pull request by minesweeter**

### CODE REVIEW

File: tutorials/alcf/02_ThetaGPU_ray/README.rst [hunk 1]
```diff
@@ -54,6 +54,7 @@ Add the following content:
     #COBALT -n 2
     #COBALT -q full-node
     #COBALT -t 20
+    #COBALT --attrs filesystems=home,grand,eagle,theta-fs0
```
Comment (Deathn0t) [inline tutorials/alcf/02_ThetaGPU_ray/README.rst:57]: add a note/tip about this, say that deephyper only requires `theta-fs0, home` + whatever directory is used by the user

File: tutorials/alcf/02_ThetaGPU_ray/README.rst [hunk 2]
```diff
@@ -124,21 +116,13 @@ Edit the ``#COBALT ...`` directives:
...
-        deephyper nas random \
-        --problem deephyper.benchmark.nas.linearRegHybrid.Problem \
-        --evaluator ray \
-        --run-function deephyper.nas.run.run_base_trainer \
-        --num-workers -1 \
-        --ray-address auto \
-        --ray-num-cpus-per-task 1 \
-        --ray-num-gpus-per-task 1 \
-        --verbose 1"
+    myscript.py
```
Comment (Deathn0t) [inline tutorials/alcf/02_ThetaGPU_ray/README.rst:125]: give an example script with using ray evaluator and 1 GPU per black-box call

File: tutorials/alcf/02_ThetaGPU_ray/README.rst [hunk 3]
```diff
@@ -124,21 +116,13 @@ Edit the ``#COBALT ...`` directives:
...
 
 Finally, submit the script from a ThetaGPU login node (e.g., ``thetagpusn1``):
```
Comment (Deathn0t) [inline tutorials/alcf/02_ThetaGPU_ray/README.rst:127]: outdated, now we have to use `qsub-gpu` also mentionne that for the installation we need to request a `single-gpu` node (or add this in installation for ThetaGPU)

File: tutorials/alcf/03_ThetaGPU_mpi/README.rst [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    # User Configuration
+    EXP_DIR=$PWD
+    INIT_SCRIPT=$PWD/init-dh-environment.sh
+    COBALT_JOBSIZE=2
+    RANKS_PER_NODE=8
+
+    # Initialization of environment
+    source $INIT_SCRIPT
+
+    echo "mpirun -x LD_LIBRARY_PATH -x PYTHONPATH -x PATH -n $(( $COBALT_JOBSIZE * $RANKS_PER_NODE )) -N $RANKS_PER_NODE --hostfile $COBALT_NODEFILE python myscript.py";
```
Comment (Deathn0t) [inline tutorials/alcf/03_ThetaGPU_mpi/README.rst:69]: remove the echo statement

File: tutorials/alcf/03_ThetaGPU_mpi/README.rst [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    #COBALT -A $PROJECT_NAME
+    #COBALT -n 2
+    #COBALT -q full-node
+    #COBALT -t 20
+    #COBALT --attrs filesystems=home,grand,eagle,theta-fs0
+
+    # User Configuration
+    EXP_DIR=$PWD
+    INIT_SCRIPT=$PWD/init-dh-environment.sh
+    COBALT_JOBSIZE=2
```
Comment (Deathn0t) [inline tutorials/alcf/03_ThetaGPU_mpi/README.rst:63]: explain why we have to define `COBALT_JOBSIZE` (not the case on Theta)

File: tutorials/alcf/03_ThetaGPU_mpi/README.rst [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    RANKS_PER_NODE=8
+
+    # Initialization of environment
+    source $INIT_SCRIPT
+
+    echo "mpirun -x LD_LIBRARY_PATH -x PYTHONPATH -x PATH -n $(( $COBALT_JOBSIZE * $RANKS_PER_NODE )) -N $RANKS_PER_NODE --hostfile $COBALT_NODEFILE python myscript.py";
+    mpirun -x LD_LIBRARY_PATH -x PYTHONPATH -x PATH -n $(( $COBALT_JOBSIZE * $RANKS_PER_NODE )) -N $RANKS_PER_NODE --hostfile $COBALT_NODEFILE python myscript.py
+
+
+Edit the ``#COBALT ...`` directives:
```
Comment (Deathn0t) [inline tutorials/alcf/03_ThetaGPU_mpi/README.rst:73]: edit with `... what ...` ?

File: tutorials/alcf/03_ThetaGPU_mpi/README.rst [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+.. code-block:: bash
+
+    #COBALT -A $PROJECT_NAME
+    #COBALT -n 2
+    #COBALT -q full-node
+    #COBALT -t 20
+    #COBALT --attrs filesystems=home,grand,eagle,theta-fs0
+
+Along with the ``COBALT_JOBSIZE`` which should always correspond to the ``-n`` attribute from the ``#COBALT ...`` directives.
```
Comment (Deathn0t) [inline tutorials/alcf/03_ThetaGPU_mpi/README.rst:83]: explained here, but explain Cleary that $PROJECT_NAME, COBALT_SIZE needs to be adapted and how (give example, or explicit current example)

File: tutorials/alcf/03_ThetaGPU_mpi/README.rst [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+And adapt the executed Python application depending on your needs:
+
+.. code-block:: python
+
+    myscript.py
```
Comment (Deathn0t) [inline tutorials/alcf/03_ThetaGPU_mpi/README.rst:89]: give example script with CBO

File: tutorials/alcf/04_ThetaGPU_jupyter/README.rst [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+Execution on the ThetaGPU supercomputer (within a Jupyter notebook)
+*******************************************************************
+
+In this tutorial we are going to learn how to run an interactive Jupyter notebook on the **ThetaGPU** supercomputer at the ALCF using Ray. `ThetaGPU <https://www.alcf.anl.gov/support-center/theta/theta-thetagpu-overview>`_ is a 3.9 petaflops system based on NVIDIA DGX A100.
+
+After logging in Theta:
+
+1. From a ``thetalogin`` node: ``ssh thetagpusn1`` to login to a ThetaGPU service node.
+2. From ``thetagpusn1``, start an interactive job (**note** which ``thetagpuXX`` node you get placed onto will vary) by replacing your ``$PROJECT_NAME`` and ``$QUEUE_NAME`` (e.g. of available queues are ``full-node`` and ``single-gpu``):
```
Comment (Deathn0t) [inline tutorials/alcf/04_ThetaGPU_jupyter/README.rst:11]: outdated no access to `thetagpusn1` anymore, we can remove `thetagpusn1`

**Review by Deathn0t — COMMENTED** [review]
I left some comments, please take them into consideration

File: tutorials/alcf/02_ThetaGPU_ray/README.rst (hunk 3 above)
Comment (minesweeter) [reply tutorials/alcf/02_ThetaGPU_ray/README.rst:127]: I wasn't aware of this ```qsub-gpu`` command

File: tutorials/alcf/03_ThetaGPU_mpi/README.rst (hunk 6 above)
Comment (minesweeter) [reply tutorials/alcf/03_ThetaGPU_mpi/README.rst:73]: I don't understand what should be done ?

**Comment by minesweeter** [comment]
To test:

- [ ] ThetaGPU (using Ray)
- [ ] ThetaGPU (using MPI)
- [ ] ThetaGPU (with Jupyter)

To do:

- [ ] specify the easy steps with Ray
     - [ ] 1 gpu
     - [ ] multi gpus
     - [ ] multi nodes (with multi gpus)

**Review by Deathn0t — APPROVED** [review]

**Pull request merged**